engine/base/runtime/src/main/scala/pl/touk/nussknacker/engine/baseengine/metrics/SourceMetrics.scala

```diff
@@ -0,0 +1,43 @@
+package pl.touk.nussknacker.engine.baseengine.metrics
+
+import cats.data.NonEmptyList
+import pl.touk.nussknacker.engine.baseengine.api.interpreterTypes.SourceId
+import pl.touk.nussknacker.engine.util.metrics.{Gauge, InstantRateMeterWithCount, MetricIdentifier, MetricsProviderForScenario}
+
+class SourceMetrics(metricsProvider: MetricsProviderForScenario) {
+
+  private val sourceMetrics = collection.concurrent.TrieMap[SourceId, MetricsForOneSource]()
+
+  def markElement(sourceId: SourceId, elementTimestamp: Long): Unit = {
+    sourceMetrics.getOrElseUpdate(sourceId, new MetricsForOneSource(sourceId)).process(elementTimestamp)
+  }
+
+  protected def now(): Long = {
+    System.currentTimeMillis()
```

Comment:
Can we just pass a `Clock` as a field to constructor? Then will be available fixed clock and won't be needed inheritance.

Reply:
:+1: I forgot about it :)